bindings/rust/s2n-tls-tokio/src/lib.rs

```diff
@@ -126,9 +126,9 @@ where
             Poll::Ready(Ok(len)) => len as c_int,
             Poll::Pending => {
                 set_errno(Errno(libc::EWOULDBLOCK));
-                s2n_status_code::FAILURE
+                CallbackResult::Failure.into()
```

Comment:
this seems alittle odd since we dont use the Success type and instead return `len as c_int`

Reply:
also wondering if CallbackResult should instead be called StatusCode so that its reusable for non callback types

Reply:
I'm not trying to solve callbacks here-- that's for later ;) I'm just trying to hide the C types.

> just an observation: this seems alittle odd since we dont use the Success type and instead return len as c_int

The interface of some s2n-tls callbacks is a little odd :( We can't use the Success value because success for this callback means a valid length, not a success status code. 

> it would be nice to have poll_io return CallbackResult here and have the downstream function handle the conversion.

There isn't really a downstream function here to handle the conversion-- this is a C callback that will return to C code that doesn't understand Rust enums.

> also wondering if CallbackResult should instead be called StatusCode so that its reusable for non callback types

I choose "CallbackResult" because a customer's only interaction with this enum should be writing callbacks, since we handle the status codes returned by s2n-tls APIs. And I would love to remove that interaction too xD